### ISSUE TITLE
Discuss non-framework libraries

### ISSUE THREAD

**Issue by Morwenn** [LoungeCPP/loungesome-cpp#13]
We still don't have any discussion for libraries too small to be considered frameworks. The kind of utility libraries that are often small, header-only, and dedicated to solving specific tasks. Without too much thought about it, I would nominate [cppformat](https://github.com/cppformat/cppformat) which seems more than decent.

**Comment by nabijaczleweli** [comment]
:+1: for `cppformat`

**Comment by ghost** [comment]
butt boost::format?

**Comment by nabijaczleweli** [comment]
@kbok [See `cppformat`'s Boost.Format README section](https://github.com/cppformat/cppformat#boost-format-library)

**Comment by nabijaczleweli** [comment]
I also propose [CPR](https://github.com/whoshuu/cpr) - a C++14-isation of `libcurl`

**Comment by ghost** [comment]
I agree with CPR - it looks nice. Let's decide whether we put CPR or [casablanca](https://casablanca.codeplex.com/) - obviously we won't just put two competing libraries into the list

**Comment by nabijaczleweli** [comment]
They are not competing, looks like.

**Comment by ghost** [comment]
Actually, it's not obvious. let's just say I'd rather not put two competing libraries, and that at some point we should decide which is awesomer. People look at this list to pick a library, so if you leave two of them, then they still have to continue searching. Just my 0.02$

**Comment by ghost** [comment]
@nabijaczleweli ok, boost::format is slow, but we still need to agree that cppformat is better :)

**Comment by nabijaczleweli** [comment]
Casablanca:

> Features:
> - HTTP client/server, JSON, URI, asynchronous streams, WebSockets client, oAuth
> - PPL Tasks - A powerful model for composing asynchronous operations based on C++ 11 features
> - Support for Visual Studio 2012, 2013, and 2015 with debugger visualizers

A.k.a. it's (a) a framework, and (b) an all-container.

CPR, however is just a C++14 wrapper around `libcurl`.

**Comment by milleniumbug** [comment]
:+1: for cppformat.

**Comment by milleniumbug** [comment]
I nominate Niebler's ranges library.

**Comment by Morwenn** [comment]
A list of the small yet cool libraries that I starred and that might be worthy of official awesomeness:
- Howard Hinnant's [date and time](https://github.com/HowardHinnant/date) library based on `<chrono>`.
- [sqlpp11](https://github.com/rbock/sqlpp11) which seems pretty cool. It implements a EDSL to manipulate databases and can even consider standard collections as temporary databases. Pretty powerful.
- Eric Niebler's range-v3 as already mentioned.
- I wish I could nominate [boost-tree](https://github.com/grafikrobot/boost-tree) which isn't part of Boost but the subject of a [standard proposal](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3700.html). It offers interesting abstractions and several kinds of generic tree data structures. IIRC it wasn't accepted in the standard library because it was too big without enough experience from users. I think it was proposed to Boost too but it overlapped with several Boost modules; I don't know what happened afterwise.
- [expected](https://github.com/ptal/expected) which is an implementation of the proposed `std::expected` error reporting utility.

All of them are "modern" C++ libraries and much thought was put in their design.

**Comment by nabijaczleweli** [comment]
:+1: the date library collection and sqlpp11
v0v expected
:-1: Boost.Tree - old, unmaintained

**Comment by nabijaczleweli** [comment]
@milleniumbug's [seed11](https://github.com/milleniumbug/seed11)

**Comment by ghost** [comment]
[pugixml](http://pugixml.org) ?

**Comment by milleniumbug** [comment]
Posting here so I won't forget: [`docopt`](https://github.com/docopt/docopt.cpp). Not yet nominating, I have to review it first.

**Comment by nabijaczleweli** [comment]
I hereby propose:
- [jsonpp](http://rapptz.github.io/jsonpp) — excellent, yet simple API
- [cereal](http://uscilab.github.io/cereal) — all-round dope serialisation

@milleniumbug, how's docopt treating yeh?
